tree/treeplayer/src/TTreeProcessorMT.cxx

```diff
@@ -192,9 +192,7 @@ MakeClusters(const std::vector<std::string> &treeNames, const std::vector<std::s
       // If the number of clusters is less than maxTasksPerFile
       // we take the clusters as they are
       if (nFolds == 0) {
-         std::for_each(
-            clustersPerFileIt->begin(), clustersPerFileIt->end(),
-            [&eventRangesPerFileIt](const EntryCluster &clust) { eventRangesPerFileIt->emplace_back(clust); });
+         *eventRangesPerFileIt = *clustersPerFileIt;
```

Comment:
I suppose this change is valid because `clustersPerFileIt` only ever points to a collection with  a single element? I.e. `clustersPerFileIt->end() - clustersPerFileIt->begin()` is guaranteed to be `1`? If so, would it make sense to assert this, to prevent future restructurings from breaking this contract?

Reply:
> clustersPerFileIt only ever points to a collection with a single element?

no I don't think that's the case.

`*clustersPerFileIt` and `*eventRangesPerFileIt` are vectors (`clusterPerFileIt` and `eventRangesPerFileIt` are iterators over `vector<vector<...>>` with same outer size which we increment in lockstep).

If I read the code correctly, here we want to do:

```
for each element in *clustersPerFileIt:
  (*eventRangesPerFileIt).emplace_back(clust);
```

i.e. add every element of `*clustersPerFileIt` to `*eventRangesPerFileIt` (which was previously empty), i.e. 

```
*eventRangesPerFileIt = *clustersPerFileIt;
```

Reply:
OK thanks. Can you move-assign?

Reply:
yep!

Reply:
done

Reply:
Nice find, that ref - I didn't see that...